src/mkdocstrings/extension.py

```diff
@@ -231,17 +231,26 @@ def _process_block(
 
 class _PostProcessor(Treeprocessor):
     def run(self, root: Element) -> None:
+        self._remove_duplicated_headings(root)
+
+    def _remove_duplicated_headings(self, parent: Element) -> bool:
         carry_text = ""
-        for el in reversed(root):  # Reversed mainly for the ability to mutate during iteration.
+        found = False
+        for el in reversed(parent):  # Reversed mainly for the ability to mutate during iteration.
             if el.tag == "div" and el.get("class") == "mkdocstrings":
                 # Delete the duplicated headings along with their container, but keep the text (i.e. the actual HTML).
                 carry_text = (el.text or "") + carry_text
-                root.remove(el)
+                parent.remove(el)
+                found = True
             elif carry_text:
                 el.tail = (el.tail or "") + carry_text
                 carry_text = ""
+            elif self._remove_duplicated_headings(el):
```

Comment:
This has a wrong behavior: "If text was carried, don't stepping into the current element". There is no connection between the two, so it shouldn't be an `else`.